tests/components/monzo/test_services.py

```diff
@@ -0,0 +1,328 @@
+"""Tests for the Monzo component."""
+
+from typing import Any
+from unittest.mock import AsyncMock
+
+import pytest
+
+from homeassistant.components.monzo.const import DOMAIN
+from homeassistant.components.monzo.sensor import MonzoSensorEntityDescription
+from homeassistant.components.monzo.services import (
+    ATTR_AMOUNT,
+    SERVICE_POT_TRANSFER,
+    TRANSFER_ACCOUNT,
+    TRANSFER_POTS,
+    TRANSFER_TYPE,
+    TRANSFER_TYPE_DEPOSIT,
+    TRANSFER_TYPE_WITHDRAWAL,
+)
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.core import HomeAssistant, State
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from . import setup_integration
+from .conftest import TEST_ACCOUNTS, TEST_POTS
+
+from tests.common import MockConfigEntry
+
+EXPECTED_VALUE_GETTERS = {
+    "balance": lambda x: x["balance"]["balance"] / 100,
+    "total_balance": lambda x: x["balance"]["total_balance"] / 100,
+    "pot_balance": lambda x: x["balance"] / 100,
+}
+
+
+async def async_get_entity_id(
+    hass: HomeAssistant,
+    acc_id: str,
+    description: MonzoSensorEntityDescription,
+) -> str | None:
+    """Get an entity id for a user's attribute."""
+    entity_registry = er.async_get(hass)
+    unique_id = f"{acc_id}_{description.key}"
+
+    return entity_registry.async_get_entity_id(SENSOR_DOMAIN, DOMAIN, unique_id)
+
+
+def async_assert_state_equals(
+    entity_id: str,
+    state_obj: State,
+    expected: Any,
+    description: MonzoSensorEntityDescription,
+) -> None:
+    """Assert at given state matches what is expected."""
+    assert state_obj, f"Expected entity {entity_id} to exist but it did not"
+
+    assert state_obj.state == str(expected), (
+        f"Expected {expected} but was {state_obj.state} "
+        f"for measure {description.name}, {entity_id}"
+    )
+
+
+async def test_basic_deposit(
+    hass: HomeAssistant,
+    monzo: AsyncMock,
+    polling_config_entry: MockConfigEntry,
+) -> None:
+    """Test all entities."""
+    await setup_integration(hass, polling_config_entry)
+    device_registry = dr.async_get(hass)
+
+    account_id = TEST_ACCOUNTS[0]["id"]
+    account_device_id = device_registry.async_get_device(
+        identifiers={(DOMAIN, account_id)}
+    ).id
+    pot_id = TEST_POTS[0]["id"]
+    pot_device_id = device_registry.async_get_device(identifiers={(DOMAIN, pot_id)}).id
```

Comment:
this looks very similar for all the tests. Can we either have a fixture or a method here, to avoid duplicate code? Or even better, would it be possible to move to parametrized tests? E.g. by parametrizing the `TRANSFER_TYPE` and the expected method called name?